### ISSUE TITLE
Peer review for proposal – Group D

### ISSUE THREAD

**Issue by khaukhau** [clarissdev/data-visualization-project-2#6]
* Novelty of the project
  * In terms of the novelty of the approach, similar projects might already exist in the field of machine learning (ML) education or visualization. To enhance its novelty, the team could focus on the **innovative interactive modules**, apart from letting users upload dataset of their choices. For instance, the team could illustrate the impact of parameter tuning on model performance. While user adjusts the parameter on learning rate, number of hidden layers, etc., this could responsively display the interactive plots showing those adjustments.
  
* Comment on Approach
  * Technology/library used
    * The proposal mentions using R and Shiny for creating user-friendly simulation of the ML pipeline. Indeed, R Shiny is suitable for building interactive web application in R. However, it is worth reconsidering whether R Shiny is optimal choice for all aspects of the project. For instance, you could explore if integrating other lib/tech alongside R Shiny could enhance the platform capabilities or performance. In addition, ensure the chosen technology stack aligns with the project's scalability and maintainability.
  * Scalability and flexibility
    * Our group wants to know the scalability and flexibility of your platform to accommodate future updates and additional features. As ML evolves, how would you ensure that your platform remains adaptable to incorporate new algorithms, datasets, and education resources.
  
* Suggestion on new features
  * Our team suggests that the team could add the **model interpretation feature** (i.e. variables explanation, plot interpretation, discussion on how that type of ML model makes prediction). Hence, this could provide insights for non-specialists get more insight to model behavior.
  * Since your project aims to make ML accessible, would incorporating features such as **interactive tutorials, explanations of ML concepts, and real-world examples** enhance learning outcomes for users at all levels of expertise.
* Question
  * The proposal provides a clear overview of high-level goals, objectives, and intended outcomes. However, there could be beneficial for further clarification:
    * Indeed, the proposal does mention several ML algorithms such as Linear Regression, MLP, KNN, Logistic Regression. However, it does not specify how these algorithms will be visualized or what specific aspects of their behavior will be underscored.
    * Also, optimization techniques would be applied, but we wonder how it will incorporate into visualization tool could be elaborated.

**Comment by thaohp113** [comment]
1. Comment on approach
- Scalability and flexibility: Currently, we are focusing on making ML inclusive to everyone, hence prioritizing introducing the simplest, easiest algorithms first. Therefore, we do not think of adding more complex algorithms at the moment, but we’ll sure include some latest news / blogs / tutorials / explanations, etc. that could be suitable for high level understanding. 
- Technology/lib used: We might need to use other libraries/frameworks, thank you for the suggestions!

2. Suggestion on new features
We believe hyperparameters are more important in the optimization processes and that will be a feature in visualizing optimization algorithms instead of Machine Learning algorithms.

3. Question
- We’ll work in details for each algorithm. For example, for KNN / logistic regressions, we could visualize the decision boundary in higher dimension, etc.
- For optimization techniques, we believe hyperparameters are more important in the optimization processes and that will be a feature in visualizing optimization algorithms instead of Machine Learning algorithms. For example, for any change in the hyperparameters (ex: learning rate), what difference will be made into the model’s performance,....

**Comment by tienvu95** [comment]
Thanks Kiet for the very detailed comment and Thao for the response. I don't think I have anything else to add.

I just want to reiterate that Kiet's note on model interpretation feature is very important. This is help you to achieve your main goal of making ML inclusive to everyone. So, focus on depth 1st. For the 1st few algorithms covered, you might want to explain its mechanism in details + provide additional tutorial (might be in forms of external links) to ensure that someone might get all the info they need about that algorithm from your product. 

** Note that you don't have to reply to this comment, this is just my thought on what you can do to improve your project.**
